### ISSUE TITLE
[Documentation] Needs info on how to use preact with create-react-app 

### ISSUE THREAD

**Issue by masbaehr** [preactjs/preact#2947]
[Documentation] Needs info on how to use preact with create-react-app 

https://stackoverflow.com/questions/58102009/how-to-convert-create-react-app-to-preact

Tried those methods but didnt seem to work with latest versions of preact , cra and craco or similar build tools

**Comment by marvinhagemeister** [comment]
Some of my clients use [react-app-rewired](https://github.com/timarney/react-app-rewired#readme) to get access to CRA's underlying webpack config. There you can add the necessary aliases.

**Comment by masbaehr** [comment]
```js
const {alias} = require('react-app-rewire-alias')

module.exports = function override(config) {
  return alias({
    "react": "preact/compat",
    "react-dom/test-utils": "preact/test-utils",
    "react-dom": "preact/compat",
  })(config)
}

```
This does not work for example:

Module not found: Can't resolve 'react' in........

**Comment by marvinhagemeister** [comment]
The package `react-app-rewired-alias` is not the same as `react-app-rewired`. I suggested the latter, not the former ;)

1. `npm install --save-dev react-app-rewired`
2. Create a `config-overrides.js` file in the same folder as your `package.json`:

```js
/* config-overrides.js */

module.exports = {
	webpack: (config, env) => {
		config.resolve.alias = {
		    // The next line is a temporary workaround and can be removed
		    // with the next Preact version
			"react/jsx-dev-runtime": "preact/jsx-runtime",
			react: "preact/compat",
			"react-dom/test-utils": "preact/test-utils",
			"react-dom": "preact/compat",
		};
		return config;
	},
};
```

3. Replace scripts in `package.json`

```diff
  "scripts": {
-   "start": "react-scripts start",
+   "start": "react-app-rewired start",
-   "build": "react-scripts build",
+   "build": "react-app-rewired build",
-   "test": "react-scripts test",
+   "test": "react-app-rewired test",
    "eject": "react-scripts eject"
  }
```

4. Run `npm start` - enjoy!

**Comment by masbaehr** [comment]
Ah it works, I was missing react/jsx-dev-runtime ! 👍 Thx

**Comment by marvinhagemeister** [comment]
@masbaehr Just released the mentioned new version of Preact `10.5.11`. With that the `react/jsx-dev-runtime` alias is not needed anymore.
